### ISSUE TITLE
Clicking menu title changes pages

### ISSUE THREAD

**Issue by jsha** [rust-lang/docs.rs#1077]
Steps to reproduce:

1. Visit a crate page, like https://docs.rs/ureq/1.5.1/ureq/.
2. Hover over the second item in the top bar, that has the crate name and version ("ureq-1.5.1").
3. A menu pops up on hover.
4. Click the second item in the top bar ("ureq-1.5.1").

Expected result:

Menu goes away.

Actual result:

Page navigates to a different page, https://docs.rs/crate/ureq/1.5.1.

Note that this problem is worse on touchscreen-based devices, since the user has to tap to open the menu in the first place, so they would reasonably expect that tapping in the same place would make the menu go away again.

**Comment by jyn514** [comment]
The menu taking you to a separate page is intended behavior - where else would you suggest to put the link? FWIW approximately the same information is available in the dropdown and on the new page.

It's strange that the menu opens for you on click, on my android it redirects me immediately without opening the dropdown.

**Comment by Nemo157** [comment]
IIRC iOS Safari has special handling to support `:hover` elements on first click.

**Comment by jsha** [comment]
Yes, I should have said: The touchscreen device I tested was an iPad. On Android I see the behavior @jyn514 describes. Although actually, it flashes the menu for a moment first before navigating to the crate page.

As a user, this navigation was confusing for a few reasons. Probably the main one was that the design of the crate page is so different from the documentation page that I though I'd wound up on an older part of the site, possibly by accident.

![image](https://user-images.githubusercontent.com/220205/95359503-13bdb280-087f-11eb-914a-0768dd34765d.png)

![image](https://user-images.githubusercontent.com/220205/95359480-0b657780-087f-11eb-9cb2-6772570efcd8.png)

For example, the big grey bar switches from vertical to horizontal orientation; the Rust logo goes away; The menu bar moves from left-anchored to centered; the items on the menu change; the logo on the "docs.rs" homepage link changes.

It was also confusing because the content on the crate page looked very similar to the content on the documentation page. In the case of the crate I was looking at, the top of the crate README replicates a lot of what's at the top of the documentation. I think that's probably a common pattern: crate authors want new visitors to see certain intro information whether they came to the crate page or the documentation page.

There's also a general UX problem of how menus should work on the web, which I think is in flux right now. I found this article pretty convincing that menus should generally be onclick: https://uxmovement.com/navigation/why-hover-menus-do-users-more-harm-than-good/. After reading that, I switched some sites I maintain from hover to onclick.

As far as where the link should go: Right now, the menu has a section "Links" that includes a link to the crate page on crates.io. I think it would make sense to put a link to docs.rs' crate page next to that, since they're mostly the same information from two different sources.

Thanks for looking at this! And thanks for all your work on docs.rs. It's an amazing tool, and the observations above are offered in the spirit of making it better, not criticizing.

**Comment by jyn514** [comment]
> As far as where the link should go: Right now, the menu has a section "Links" that includes a link to the crate page on crates.io. I think it would make sense to put a link to docs.rs' crate page next to that, since they're mostly the same information from two different sources.

Oh hmm, that's not a bad idea. We could also make 'UREQ 1.5.0' clickable so the UI is mostly the same as before.

> It was also confusing because the content on the crate page looked very similar to the content on the documentation page. In the case of the crate I was looking at, the top of the crate README replicates a lot of what's at the top of the documentation. I think that's probably a common pattern: crate authors want new visitors to see certain intro information whether they came to the crate page or the documentation page.

Right, I don't think there's much we can do about that. We've actually had feature requests to make them *more* similar by showing the README at the top of the rustdoc page if there are no top-level docs (probably not going to happen because it requires rustdoc support).

> For example, the big grey bar switches from vertical to horizontal orientation; the Rust logo goes away; The menu bar moves from left-anchored to centered; the items on the menu change; the logo on the "docs.rs" homepage link changes.

Yeah, the design is not ideal ... this would be helped by #918 but the layout would still be different. I think a good start would be to make it left-aligned instead of centered, would you be interested in working on that? Switching the logo to be 'three boxes' on both pages would be a good change too I think. The other things need more design.

> Thanks for looking at this! And thanks for all your work on docs.rs. It's an amazing tool, and the observations above are offered in the spirit of making it better, not criticizing.

❤️ ❤️ ❤️

**Comment by jyn514** [comment]
The things left are
- make 'UREQ 1.5.0' clickable
- the big grey bar switches from vertical to horizontal orientation
- the Rust logo goes away
- The menu bar moves

> the items on the menu change

I'm not sure what we can do about this ... the information on the pages is different. On the /crate page, it's about the _crate_, on the main page it's abaout the _documentation_.

**Comment by Nemo157** [comment]
Though, we could have the platform links on /crate in order to go directly from that page to the platform-specific documentation (currently you have to click "documentation" then choose the platform from the dropdown). I think it could be possible to have all the menu items shown on both the documentation and /crate pages (and make sure the locations of the docs.rs items stay still on the internal docs.rs pages vs documentation page).
